zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -260,10 +616,66 @@ public String login(AuthenticationType authType) {
     public String getAccountUsername() {
         HAccount authenticatedAccount =
                 ServiceLocator.instance().getInstance(
-                        JpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
+                        ZanataJpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
         if (authenticatedAccount != null) {
             return authenticatedAccount.getUsername();
         }
         return null;
     }
+
+    public Principal getPrincipal() {
+        return principal;
+    }
+
+    public void checkRestriction(String expression) {
+        log.warn("permission check as EL is not supported yet");
+        throw new UnsupportedOperationException("identity.checkRestriction(exp) is not supported");
```

Comment:
- [ ] @huangp In that case, we should get rid of the `value` parameter for `@CheckPermission` (or the whole annotation), along with methods like this one for handling it.